File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        String[] str = s.split("");
        int[] answer = new int[str.length];
        Map<String, Integer> map = new HashMap<>();
        for(int i=0; i<str.length; i++){
            answer[i] = map.get(str[i])==null? -1 : i - map.get(str[i]);
            map.put(str[i], i);
        }
        return answer;
    }
}

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    let idx = {}
    for(var i=0; i<s.length; i++){
        let ch = s[i];
        if(idx[ch] != undefined){
            answer.push(i - idx[ch]);
        }else{
            answer.push(-1);
        }
        idx[ch] = i;
    }
    return answer;
}